### PR TITLE
[posix] fix cast-align build error on ARM

### PR DESCRIPTION
**Pull request by Moandor-y**
This fixes the build error by ignoring. Fixes https://github.com/openthread/openthread/issues/5647.

### CODE REVIEW

File: src/posix/platform/netif.cpp [hunk 1]
```diff
@@ -1027,11 +1032,20 @@ static void processNetifInfoEvent(otInstance *aInstance, struct rt_msghdr *rtm)
...
     const size_t kMaxNetifEvent = 8192;
     ssize_t      length;
-    char         buffer[kMaxNetifEvent];
+
+#if defined(__linux__)
+    alignas(nlmsghdr)
+#else
+    alignas(rt_msghdr)
+#endif
+        char buffer[kMaxNetifEvent];
```
Comment (Irving-cl) [inline src/posix/platform/netif.cpp:1048]: Is this alignment done by `clang-format`?

**Review by Irving-cl — APPROVED** [review]
LGTM

File: src/posix/platform/netif.cpp (hunk 1 above)
Comment (Moandor-y) [reply src/posix/platform/netif.cpp:1048]: Yes.

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**